### PR TITLE
Add new python config loader class for state manager

### LINKED ISSUE #1343: Executor is ignoring local state manager location
This issue is related to #1313 (dbe0563)

In a test I am running YARN scheduler on a single node with local state manager (config below)
```
heron.class.state.manager:        com.twitter.heron.statemgr.localfs.LocalFileSystemStateManager
heron.statemgr.connection.string: LOCALMODE
heron.statemgr.root.path:         ${HOME}/.herondata/repository/state/${CLUSTER}
``` 

Here `CLUSTER = yarn`

The state manager is persisting topology and packing plans at the correct location, under ~/.herondata/repository/state/**yarn**. 

However the executor seems to be looking for the packing plan under a different location (~/.herondata/repository/state/**local**). Here is the relevant info from the executor logs:
```
...
09/05/2016 11:20:41 -0700:INFO: state manager configs: {'hostport': 'LOCALMODE', 'type': 'file', 'name': 'local', 'tunnelhost': 'localhost', 'rootpath': '~/.herondata/repository/state/local'}
09/05/2016 11:20:41 -0700:INFO: Connecting to file state with rootpath: /home/.herondata/repository/state/local
...
```

As a result executor fails to start any processes.

### PR DESCRIPTION
**Pull request by billonahill**
Fixes #1343 by handling wildcard substitution in config values. Also creates a new configloader class to handle loading/transforming state manager configs. 

Also changes `//heron/statemgrs/src/python:zk-statemgr-py` to remove the `zk` part since this target also includes local state managers: `//heron/statemgrs/src/python:statemgr-py`.

### CODE REVIEW

File: heron/statemgrs/src/python/configloader.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      found = re.findall(r'\${[A-Z_]+}', config_value)
+      if found:
+        raise ValueError("%s=%s in file %s contains unsupported or unset wildcard tokens: %s" %
+                         (config_key, original_value, config_file, ", ".join(found)))
+      config[config_key] = config_value
+  return config
+
+if __name__ == "__main__":
+  if len(sys.argv) > 1:
+    locations = load_state_manager_locations('local', sys.argv[1])
```
Comment (kramasamy) [inline heron/statemgrs/src/python/configloader.py]: Is this for testing - when you mentioned 'local'?

File: heron/statemgrs/src/python/configloader.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  return [state_manager_location]
+
+def __replace(config, wildcards, config_file):
+  """For each kvp in config, do wildcard substitution on the values"""
+  for config_key in config:
+    config_value = config[config_key]
+    original_value = config_value
+    if isinstance(config_value, basestring):
+      for token in wildcards:
+        print("config_value: %s, token: %s wildcard value: %s" %
```
Comment (objmagic) [inline heron/statemgrs/src/python/configloader.py]: can we use `Log` here?

Comment (billonahill) [inline heron/statemgrs/src/python/configloader.py]: This was for debugging, removed.

File: heron/statemgrs/src/python/configloader.py (hunk 1 above)
Comment (billonahill) [inline heron/statemgrs/src/python/configloader.py]: Yes, this was just for an easy way to manually check a file besides those in the unit tests.

**Comment by kramasamy** [comment]
please write a comment - that it is used for local testing only 👍

**Pull request merged**